Can't resolve 'vertx'

iv tried doing the 

`new webpack.IgnorePlugin(/vertx/)`

not working for me

```WARNING in ./node_modules/es6-promise/dist/es6-promise.js
Module not found: Error: Can't resolve 'vertx' in '/*******/node_modules/es6-promise/dist'
 @ ./node_modules/es6-promise/dist/es6-promise.js 138:16-26
 @ ./client/src/App.jsx
 @ ./client/src/routes.jsx
 @ ./client/src/index.jsx
 @ multi ./client/src/index
webpack: Compiled with warnings.```

--- a/lib/es6-promise/asap.js
+++ b/lib/es6-promise/asap.js
@@ -96,8 +96,7 @@ function flush() {
 
 function attemptVertx() {
   try {
-    const r = require;
-    const vertx = r('vertx');
+    const vertx = Function('return this')().require('vertx');
     vertxNext = vertx.runOnLoop || vertx.runOnContext;
     return useVertxTimer();
   } catch(e) {

